KarhooSDK/Network/HttpClient/TokenRefreshingHttpClient/Interactor/KarhooRefreshTokenInteractor.swift

```diff
@@ -28,15 +28,32 @@ final class KarhooRefreshTokenInteractor: RefreshTokenInteractor {
     }
 
     func refreshToken(completion: @escaping (Result<Bool>) -> Void) {
-        self.callback = completion
+        callback = completion
 
         guard tokenNeedsRefreshing() else {
             completion(Result.success(result: false))
             return
         }
 
         guard let refreshToken = dataStore.getCurrentCredentials()?.refreshToken else {
```

Comment:
the refresh token is an empty string, in which case the old flow is triggered. Please check for both nil and empty scenarios and trigger the new flow for either case.

Reply:
You are absolutely right. New condition added to guard statement. :+1: